linera-base/src/data_types.rs

```diff
@@ -718,24 +718,45 @@ pub struct OracleRecord {
     pub responses: Vec<OracleResponse>,
 }
 
+impl OracleRecord {
+    /// Wether an `OracleRecord` is permitted in fast blocks or not.
+    pub fn is_permitted_in_fast_blocks(&self) -> bool {
+        self.responses.is_empty()
+            || self
```

Comment:
`all` is `true` for an empty iterator.

```suggestion
        self
```

Reply:
Ah nice ok, good to know